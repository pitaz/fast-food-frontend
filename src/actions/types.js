export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const SIGN_UP_ERRORS = 'SIGN_UP_ERRORS';
export const REMOVE_ERROR_MESSAGE = 'REMOVE_ERROR_MESSAGE';
export const SET_CURRENT_USER_FAIL = 'SET_CURRENT_USER_FAIL';
export const LOG_OUT_USER = 'LOG_OUT_USER';
export const GET_MEALS = 'GET_MEALS ';
