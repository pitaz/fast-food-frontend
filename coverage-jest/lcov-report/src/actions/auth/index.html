<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/actions/auth</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/actions/auth
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.96% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>17/46</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.18% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>15/33</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="logout.js"><a href="logout.js.html">logout.js</a></td>
	<td data-value="33.33" class="pic low"><div class="chart"><div class="cover-fill" style="width: 33%;"></div><div class="cover-empty" style="width:67%;"></div></div></td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	</tr>

<tr>
	<td class="file low" data-value="signIn.js"><a href="signIn.js.html">signIn.js</a></td>
	<td data-value="36.84" class="pic low"><div class="chart"><div class="cover-fill" style="width: 36%;"></div><div class="cover-empty" style="width:64%;"></div></div></td>
	<td data-value="36.84" class="pct low">36.84%</td>
	<td data-value="19" class="abs low">7/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="14" class="abs low">6/14</td>
	</tr>

<tr>
	<td class="file low" data-value="signUp.js"><a href="signUp.js.html">signUp.js</a></td>
	<td data-value="38.1" class="pic low"><div class="chart"><div class="cover-fill" style="width: 38%;"></div><div class="cover-empty" style="width:62%;"></div></div></td>
	<td data-value="38.1" class="pct low">38.1%</td>
	<td data-value="21" class="abs low">8/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="10" class="abs low">2/10</td>
	<td data-value="46.67" class="pct low">46.67%</td>
	<td data-value="15" class="abs low">7/15</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 06 2019 12:21:36 GMT+0100 (West Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
