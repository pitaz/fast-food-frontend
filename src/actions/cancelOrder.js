import axios from 'axios';
import {toastr} from 'react-redux-toastr';
import * as types from './types';

export const cancelOrderSuccess = orderId => ({
  type: types.CANCEL_ORDER,
  orderId,
});
export const cancelOrderError = error => ({
  type: types.CANCEL_ORDER_FAIL,
  error,
});

const  cancelOrder = (orderId) => (dispatch) => {
	return axios.delete(`https://fast-food-pitaz.herokuapp.com/api/v1/orders/${orderId}`).then(
		(res) => {
      dispatch(cancelOrderSuccess(orderId));
      toastr.warning('Order has been cancelled');
		}
	).catch((error) => {
		dispatch(cancelOrderError(error.response));
	});
};

export default cancelOrder;
