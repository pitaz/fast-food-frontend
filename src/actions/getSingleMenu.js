import axios from 'axios';
import * as types from './types';

export const getSingleMenu = menu => ({
  type: types.GET_SINGLE_MENU,
  menu,
});
export const getSingleMenuError = errors => ({
  type: types.GET_SINGLE_MENU_FAIL,
  errors,
});

const fetchSingleMenu = (mealId) => (dispatch) => {
	return axios.get(`https://fast-food-pitaz.herokuapp.com/api/v1/menu/${mealId}`).then(
		(res) => {
      dispatch(getSingleMenu(res.data.data.item));
		}
	).catch((error) => {
		dispatch(getSingleMenuError(error.response.data));
	});
};

export default fetchSingleMenu;
