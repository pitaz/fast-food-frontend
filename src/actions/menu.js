import axios from 'axios';
import * as types from './types';

export const getMeals = menu => ({
  type: types.GET_MEALS,
  menu,
});
export const getMealsError = error => ({
  type: types.GET_MEALS,
  error,
});

const fetchMeal = () => (dispatch) => {
	return axios.get('https://fast-food-pitaz.herokuapp.com/api/v1/menu').then(
		(res) => {
      dispatch(getMeals(res.data.data.items));
		}
	).catch((error) => {
		console.log(error);
		
		dispatch(getMealsError(error.response.data));
	});
};

export default fetchMeal;
