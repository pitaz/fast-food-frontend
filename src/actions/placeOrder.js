import axios from 'axios';
import {toastr} from 'react-redux-toastr';
import { PLACE_ORDER_SUCCESS, PLACE_ORDER_FAIL } from './types';

export const placeOderSuccess = order => ({
  type: PLACE_ORDER_SUCCESS,
  order,
});

export const placeOderFail = error => ({
  type: PLACE_ORDER_FAIL,
  error,
});

const placeOrder = (payload, history) => dispatch => {
  let message = '';
  return axios.post('https://fast-food-pitaz.herokuapp.com/api/v1/orders', payload).then(
		(res) => {
      dispatch(placeOderSuccess(res.data.data));
      toastr.success('Success', res.data.message);
      history.push('/menu');
		}
	).catch((error) => {
    if (error.response.status === 401) message = 'You must be logged in to place an order';
    if (error.response.status === 500) message = 'Oops something went wrong, try again!';
    dispatch(placeOderFail(error.response.data.message));
    toastr.error('Operation unsuccessful', message);
	});
};

export default placeOrder;
