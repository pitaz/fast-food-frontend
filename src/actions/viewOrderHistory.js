import axios from 'axios';
import * as types from './types';

export const getOrderHistory = orders => ({
  type: types.FETCH_ORDER_HISTORY,
  orders,
});
export const getOrderHistoryError = errors => ({
  type: types.FETCH_ORDER_HISTORY_FAIL,
  errors,
});

const fetchOrderHistory = () => (dispatch) => {
  
  return axios.get('https://fast-food-pitaz.herokuapp.com/api/v1/users/1/orders').then(
    (res) => {
      dispatch(getOrderHistory(res.data.data.items));
		}
	).catch((error) => {
		dispatch(getOrderHistoryError(error.response.message));
	});
};

export default fetchOrderHistory;
