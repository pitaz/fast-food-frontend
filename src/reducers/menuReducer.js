import * as types from "../actions/types";

const initialState = {
  menu: [],
  singleMenu: null
};

export default (state = initialState, action) => {
  switch (action.type) {
    case types.GET_MEALS:
      return {
        ...state,
        menu: action.menu
      };
			default: return state;
  }
};
