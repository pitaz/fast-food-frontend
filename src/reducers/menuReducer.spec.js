/* eslint-disable no-undef */
import * as types from '../actions/types';
import menuReducer from './menuReducer';

describe('select_reducer', () => {
  it('renders with initial state', () => {
    const action = { type: types.GET_MEALS };
    const initialState = { error: {}, menu: undefined, singleMenu: null };
    expect(menuReducer(undefined, action)).toEqual(initialState);
  });

  it('renders correctly', () => {
    const action = { type: types.GET_MEALS };
    expect(menuReducer(undefined, action)).toMatchSnapshot();
  });

  it('returns the correct state', () => {
    const action = { type: types.GET_MEALS, menu: [] };
    expect(menuReducer(undefined, action)).toMatchSnapshot();
  });
});
