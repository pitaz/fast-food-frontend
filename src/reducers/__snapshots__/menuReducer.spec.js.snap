// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_reducer renders correctly 1`] = `
Object {
  "menu": undefined,
  "singleMenu": null,
}
`;

exports[`select_reducer returns the correct state 1`] = `
Object {
  "menu": Array [],
  "singleMenu": null,
}
`;
